### PR TITLE
Auto comp

### PR DESCRIPTION
**Pull request by Juxtaposedwords**
Adding auto-completion to commands

### CODE REVIEW

File: out/automation/util/early-game-start-hack.js [hunk 1]
```diff
@@ -52,4 +52,21 @@ export async function main(ns) {
...
+export function autocomplete(data, args) {
+	data.flags([
+		["server", null],  // the server that will run the scripts
+		["target", null],  // the target server
+		["verbose", false], // print errors if launching the scripts fails
+	])
+	if (args.length === 1 || args.length === 2){
+		return data.servers
+	}
+	const options = {
```
Comment (uhhhclem) [inline out/automation/util/early-game-start-hack.js:66]: How does declaring this constant affect anything?

File: out/automation/util/start-hack.js [hunk 2]
```diff
@@ -61,3 +61,21 @@ export async function main(ns) {
...
+		["server", null],  // the server that will run the scripts
+		["target", null],  // the target server
+		["verbose", false], // print errors if launching the scripts fails
+	])
+	const options = {
+		"server": data.servers,  // the server that will run the scripts
+		"target": data.servers,  // the target server
+	}
+
+	for (let arg of args.slice(-2)) {
```
Comment (uhhhclem) [inline out/automation/util/start-hack.js:75]: I suspect this code needs to be present in the autocomplete function for early-game-start-hack.js.

File: out/startup.js [hunk 3]
```diff
@@ -59,3 +59,10 @@ export async function main(ns) {
         await ns.sleep(60000)
     }
 }
+
+export function autocomplete(data, args) {
```
Comment (uhhhclem) [inline out/startup.js:63]: You don't need this.  Boolean flags don't have values, they're just true if present on the command line and false if not. There's no need to autocomplete them.

File: out/automation/util/early-game-start-hack.js (hunk 1 above)
Comment (Juxtaposedwords) [reply out/automation/util/early-game-start-hack.js:66]: This ensures the first or second argument passed will autocomplete out of all server names.

File: out/startup.js (hunk 3 above)
Comment (Juxtaposedwords) [reply out/startup.js:63]: This isn't about values, but the flag name. Flags are not autocomplete unless there data.flags is called

File: out/automation/util/start-hack.js (hunk 2 above)
Comment (Juxtaposedwords) [reply out/automation/util/start-hack.js:75]: As ever, you have the right of it.

File: out/startup.js (hunk 3 above)
Comment (Juxtaposedwords) [reply out/startup.js:63]: " Calling this function adds all the flags as autocomplete arguments"
https://bitburner.readthedocs.io/en/latest/netscript/advancedfunctions/autocomplete.html

File: out/automation/util/early-game-start-hack.js (hunk 1 above)
Comment (Juxtaposedwords) [reply out/automation/util/early-game-start-hack.js]: it does not! good call. I'd gotten mixed up, as early-game has no flags.

**Pull request merged**